overlord/devicestate/handlers.go

```diff
@@ -189,3 +200,47 @@ func (m *DeviceManager) doMarkSeeded(t *state.Task, _ *tomb.Tomb) error {
 	st.EnsureBefore(0)
 	return nil
 }
+
+// preseedFilePatterns contains files/globs that should be excluded/included in the
+// preseeding artifact created by snap-preseed command for uc20 system.
+var preseedFilePatterns = struct {
+	Exclude []string `json:"exclude"`
+	Include []string `json:"include"`
+}{
+	Exclude: []string{
+		"var/lib/snapd/snaps/*.snap",
+		"var/lib/seed/*",
+	},
+	Include: []string{
+		"etc/udev/rules.d",
+		"etc/systemd",
+		"etc/dbus-1",
+		"snap",
+		"var/lib/snapd/state.json",
+		"var/lib/snapd/apparmor",
+		"var/lib/snapd/system-key",
+		"var/lib/snapd/assertions",
+		"var/lib/snapd/seccomp",
+		"var/lib/snapd/desktop",
+		"var/lib/snapd/sequence",
+		"var/lib/snapd/cookie",
+		"var/lib/snapd/dbus-1",
+		"var/snap",
+		"var/cache/snapd/aux",
+		"var/cache/apparmor",
+	},
+}
+
+// dumpPreseedFilePatterns stores preseedFilePatterns in /var/lib/snapd/preseed-export.json file;
+// the file is consumed by snap-preseed command when creating preseed.tgz artifact for UC20.
+func dumpPreseedFilePatterns() error {
+	data, err := json.Marshal(preseedFilePatterns)
+	if err != nil {
+		return err
+	}
+
+	if err := ioutil.WriteFile(filepath.Join(dirs.SnapdStateDir(dirs.GlobalRootDir), "preseed-export.json"), data, 0644); err != nil {
```

Comment:
nitpick: maybe we should put this into /var/cache/snapd or somewhere in /tmp?

Reply:
also there is nothing dynamic in it? do you think it will grow dynamic bits? couldn't it be a static file inside the snapd snap?

Reply:
@pedronis it could be a static file but my motivation to generate it instead was that it's all too easy to modify it if it's a static one. WDYT?

Reply:
I'm not sure I understand the concern. We would read it out of the snapd snap which we control as we use the snapd binary from it.  I would create a preseed.json  parallel to the info file I think.

Reply:
yes